src/engine/bank04.asm

```diff
@@ -1180,8 +1180,48 @@ Func_12871: ; 12871 (4:6871)
 Func_1288c: ; 1288c (4:688c)
 	INCROM $1288c, $128a9
 
-Func_128a9: ; 128a9 (4:68a9)
-	INCROM $128a9, $1296e
+OnPlayerNamingScreen:: ; 128a9 (4:68a9)
+	; clear name buffer.
+	ld hl, wc500 ; c500: name buffer.
+	ld bc, $0010
+	ld a, $00
+	call FillMemoryWithA
+
+	; get player's name
+	; from the user into hl.
+	ld hl, wc500
+	farcall InputPlayerName
+	
+	farcall Func_c1a4
+	call DoFrameIfLCDEnabled
+	call DisableLCD
+	ld hl, wc500
+	; get the first byte of the name buffer.
+	ld a, [hl]
+	or a
+	; check if anything typed.
+	jr nz, .no_name
+	ld hl, .data
+.no_name
+	; set the default name.
+	ld de, sPlayerName
+	ld bc, $0010
```

Comment:
A constant could be appropriate here (e.g. ``PLAYER_NAME_SIZE``), and it could also replace ``https://github.com/pret/poketcg/blob/master/src/sram.asm#L26`` and ``https://github.com/pret/poketcg/blob/master/src/wram.asm#L367``.